draft-ramseyer-grow-peering-api.md

```diff
@@ -350,11 +290,96 @@ The client will send a structure along with the request, as follows:
 
 The server then responds with the same structure, with the information that it understands (status, etc).
 
-COMPLETION      {#completion}
-----------
+### COMPLETION      {#completion}
 If both sides report that the session is established, then peering is complete.
 If one side does not configure sessions within the server's acceptable configuration window (TimeWindow), then the server is entitled to remove the configured sessions and report "Unestablished" to the client.
 
+Authentication
+==============
+
+Authentication to this API is performed with Bearer tokens. Two methods are defined below to obtain a bearer token,
+the choice of which is supported is up to operator preference however the authors recommend the use of RPKI Signed
+Checklists over reliance on an external OIDC provider.
+
+OAuth with RPKI Signed Checklists
+---------------------------------
+
+This section of the document defined an extension to the OAuth 2.0 protocol ({{!RFC6749}}) to allow for a client to
+obtain an access token using RPKI Signed Checklists ({{!RFC9323}}).
+
+### Server Metadata
+
+A server implementing OAuth with RSCs MUST host an OAuth Authorization Server Metadata ({{!RFC8414}}) at
+`/.well-known/oauth-authorization-server`. That is if its Peering API base URL is `https://example.com/peering`
+then its OAuth Metata file will be at `https://example.com/peering/.well-known/oauth-authorization-server`.
+
+The metadata file MUST include the following additional field:
+
+* `rpki_signed_checklist_nonce`: a URL the client can use to obtain a nonce for the OAuth Exchange.
+
+### Nonce Endpoint
+
+The OAuth Server's `rpki_signed_checklist_nonce` endpoint MUST accept GET requests and MUST return a JSON document with
+two fields:
+
+* `state`: an opaque string to be returned to the server to allow it to keep track of state
+* `nonce`: a Base64 encoded nonce containing at least 128 bits of entropy
+
+The server SHOULD set a reasonable time limit on when the returned nonce can be used.
+
+### Token Endpoint
+
+The standard OAuth token endpoint is extended with a new `grant_type` of
+`urn:ietf:params:oauth:grant-type:rpki_signed_checklist`. This grant type has two fields:
+
+* `state`: the opaque state string associated with the nonce
+* `rsc`: a Base64 encoded RPKI Signed Checklist over the ASNs the client wishes to be authorized for
+
+### RPKI Signed Checklist Format
```

Comment:
I'd replace all instances of `RSC` with `RSM`

Reply:
Done.